QuizApp/QuizApp/Client/NetworkClient.swift

```diff
@@ -0,0 +1,172 @@
+import Foundation
+
+protocol NetworkClientProtocol {
+
+    func executeRequest(path: String) async throws
+
+    func executeRequest<T: Decodable, E: Encodable>(
+        path: String,
+        method: RequestMethod,
+        header: [String: String]?,
+        body: E) async throws -> T
+
+    func executeRequest<T: Decodable, E: Encodable>(path: String, method: RequestMethod, body: E) async throws -> T
+
+    func executeRequest<T: Decodable>(path: String, method: RequestMethod) async throws -> T
+
+}
+
+class NetworkClient: NetworkClientProtocol {
+
+    private let baseUrl: String
+    private let securityStorage: SecurityStorageProtocol
+
+    init(baseUrl: String, securityStorage: SecurityStorageProtocol) {
+        self.baseUrl = baseUrl
+        self.securityStorage = securityStorage
+    }
+
+    func executeRequest(path: String) async throws {
+        guard let token = securityStorage.accessToken else {
```

Comment:
Why are you throwing error here? If the token does not exist, that means the server will return 401 and a more correct error will be thrown.